src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -1032,4 +1053,144 @@ private StringConcatFactory() {
         // no instantiation
     }
 
+    /**
+     * Bytecode StringBuilder strategy.
+     *
+     * <p>This strategy emits StringBuilder chains as similar as possible
+     * to what javac would. No exact sizing of parameters or estimates.
+     */
+    private static final class SimpleStringBuilderStrategy {
+        static final String METHOD_NAME = "concat";
+        static final ClassDesc STRING_BUILDER = ClassDesc.ofDescriptor("Ljava/lang/StringBuilder;");
+        static final ClassFileDumper DUMPER;
+        static final MethodTypeDesc APPEND_BOOLEAN_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_boolean);
+        static final MethodTypeDesc APPEND_CHAR_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_char);
+        static final MethodTypeDesc APPEND_DOUBLE_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_double);
+        static final MethodTypeDesc APPEND_FLOAT_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_float);
+        static final MethodTypeDesc APPEND_INT_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_int);
+        static final MethodTypeDesc APPEND_LONG_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_long);
+        static final MethodTypeDesc APPEND_OBJECT_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_Object);
+        static final MethodTypeDesc APPEND_STRING_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_String);
+        static final MethodTypeDesc INT_VOID_TYPE = MethodTypeDesc.of(ConstantDescs.CD_void, ConstantDescs.CD_int);
+
+        /**
+         * Ensure a capacity in the initial StringBuilder to accommodate all constants plus this factor times the number
+         * of arguments.
+         */
+        static final int ARGUMENT_SIZE_FACTOR = 4;
+
+        public static final Set<Lookup.ClassOption> SET_OF_STRONG = Set.of(STRONG);
+
+        static {
+            DUMPER = ClassFileDumper.getInstance("java.lang.invoke.StringConcatFactory.dump", "stringConcatClasses");
```

Comment:
Nit: this static block isn't strictly needed.  Can assign at the declaration of the static variable.